app/services/foreman/parameters/validator.rb

```diff
@@ -0,0 +1,54 @@
+module Foreman
+  module Parameters
+    class Validator
+      KEY_DELM = ","
+
+      def initialize(item, options = {})
+        @item, @options = item, options
+      end
+
+      def validate!
+        case @options[:type].to_s
+        when "regexp"
+          validate_regexp
+        when "list", "array"
+          validate_list
+        else
+          return true
+        end
+      end
+
+      private
+
+      def value
+        @item.send(@options[:getter])
+      end
+
+      def validate_regexp
+        unless (/#{@options[:validate_with]}/) =~ value
```

Comment:
This causes an error with non-string data types, e.g. an integer param with a regexp validator throws `no implicit conversion of Fixnum into String`.  The original code, the other way around (`value =~ /../`) works correctly.  The brackets are also superfluous.